usbsdmux/usb2642i2c.py

```diff
@@ -361,24 +359,15 @@ def write_config(self, data):
                                        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,\
                                        0x00, 0x00, 0x00], 16)
 
-    # First looked like the data block was prefixed with some magic.
-    # But at a second look this was not the case. Setting the length of this
-    # field to 0 bytes prevents any padding.
-    data_prefix = list_to_uint8_array([], 0)
-
     # Data in the captured USB-transfer was suffixed with some random data.
     # Experiments showed that 0x00 works fine too.
     # Since the buffer is zero-ed when initialized the suffix could be removed.
     data_suffix = list_to_uint8_array([0x00], 127)
 
     # Copying prefix, data and suffix to the SCSI command data-section
     payload = (ctypes.c_uint8*512)()
-    for i in range(ctypes.sizeof(data_prefix)):
-      payload[i] = data_prefix[i]
-    for i in range(ctypes.sizeof(data)):
-      payload[i+ctypes.sizeof(data_prefix)] = data[i]
-    for i in range(ctypes.sizeof(data_suffix)):
-      payload[i+ctypes.sizeof(data_prefix)+ctypes.sizeof(data)] = data_suffix[i]
+    payload[:ctypes.sizeof(data)] = data
+    payload[ctypes.sizeof(data):ctypes.sizeof(data) + ctypes.sizeof(data_suffix)] = data_suffix
```

Comment:
It looks to me like `data_suffix` contains `511` zeroes and that `payload` already contains these zeroes at this point.
I would prefer replacing the copying of `data_suffix` to `payload` with a comment describing the differences between what you see when using the Vendor tool and what is done here.
(e.g. the vendor tool adding a prefix and suffix and us leaving both of them out).

Reply:
I took another look at this change and I am fine with it.